dff/pipeline/pipeline/component.py

```diff
@@ -27,22 +32,21 @@ class PipelineComponent(abc.ABC):
     This class represents a pipeline component, which is a service or a service group.
     It contains some fields that they have in common.
 
-    :param before_handler: before handler, associated with this component
-    :param after_handler: after handler, associated with this component
-    :param timeout: (for asynchronous only!) maximum component execution time (in seconds),
-        if it exceeds this time, it is interrupted
-    :param requested_async_flag: requested asynchronous property;
-        if not defined, calculated_async_flag is used instead
-    :param calculated_async_flag: whether the component can be asynchronous or not
-
-        - for :py:class:`~pipeline.service.service.Service`: whether its ``handler`` is asynchronous or not
-        - for :py:class:`~pipeline.service.group.ServiceGroup`: whether all its ``services`` are asynchronous or not
-
-    :param start_condition: StartConditionCheckerFunction that is invoked before each component execution;
-        component is executed only if it returns True
-    :param name: component name (should be unique in single :py:class:`~pipeline.service.group.ServiceGroup`),
-        should not be blank or contain `.` symbol
-    :param path: separated by dots path to component, is universally unique.
+    :param before_handler: Before handler, associated with this component.
+    :param after_handler: After handler, associated with this component.
+    :param timeout: (for asynchronous only!) Maximum component execution time (in seconds),
+        if it exceeds this time, it is interrupted.
+    :param requested_async_flag: Requested asynchronous property;
+        if not defined, `calculated_async_flag` is used instead.
+    :param calculated_async_flag: Whether the component can be asynchronous or not
+        1) for :py:class:`~pipeline.service.service.Service`: whether its `handler` is asynchronous or not,
+        2) for :py:class:`~pipeline.service.group.ServiceGroup`: whether all its `services` are asynchronous or not.
+
+    :param start_condition: :py:class:`~StartConditionCheckerFunction` that is invoked before each component execution;
```

Comment:
Sphinx doesn't recognize this reference for two reasons:

1. `StartConditionCheckerFunction` is a type not a class, so `:py:data:` directive should be used instead.
2. This type is `dff.pipeline.types.StartConditionCheckerFunction`, not `StartConditionCheckerFunction`, so a dot should be used.
```suggestion
    :param start_condition: :py:data:`~.StartConditionCheckerFunction` that is invoked before each component execution;
```
Fix these two issues here and in other places.